public/js/javascript.js

```diff
@@ -1,24 +1,141 @@
-/**
- * Created by mockie on 11/7/14.
- */
-
-jQuery('document').ready(function(){
-
-    jQuery('a.btn-donate').click(function(){
-        var state = jQuery(this).data('toggleState');
-        if(!state){
-            jQuery(this).next('.donation-desc').slideDown();
-        } else {
-            jQuery(this).next('.donation-desc').slideUp();
-        }
-        $(this).data('toggleState', !state);
-        return false;
-    });
-
-    jQuery('ul.contributor-gallery li').hover(function(){
-        $(this).find('.gallery-overlay').stop(true).fadeIn(1500);
-    },function(){
-        $(this).find('.gallery-overlay').stop(true).fadeOut(1500);
-    });
-
-});
+/**
+ * Created by mockie on 11/7/14.
+ */
+
+jQuery('document').ready(function(){
+
+    jQuery('a.btn-donate').click(function(){
+        var state = jQuery(this).data('toggleState');
+        if(!state){
+            jQuery(this).next('.donation-desc').slideDown();
+        } else {
+            jQuery(this).next('.donation-desc').slideUp();
+        }
+        $(this).data('toggleState', !state);
+        return false;
+    });
+
+    jQuery('ul.contributor-gallery li').hover(function(){
+        $(this).find('.gallery-overlay').stop(true).fadeIn(1500);
+    },function(){
+        $(this).find('.gallery-overlay').stop(true).fadeOut(1500);
+    });
+
+});
+
+;(function (w, d, $, undefined) {
+    'use strict';
+
+    var ajaxImageUpload = {
+
+        /**
+         * Attach event listeners
+         */
+        init: function () {
+
+            $("button.upload").on("click", function () {
+                $(".uploader-inline").show();
+                $(".gallery-view").hide().find("figure.centered").remove();
+            });
+
+            $(".gallery").on("click", function() {
+                ajaxImageUpload.showFiles();
+            });
+
+            $("button.media-modal-toggle").on("click", function () {
+                $("#modal-imgupload").fadeToggle(850);
+            });
+
+            /**
+             * AJAX image upload
+             */
+            $("#imgajax").change(function (e) {
+                var $form = $("#content").submit();
+                $.ajax({
+                    url: $form.attr("action"),
+                    type: "POST",
+                    data: new FormData($form[0]),
+                    processData:false,
+                    contentType:  false,
+                    cache: false,

```

Comment:
we need <code>accepts: "application/json"</code> to be handled by <code>acceptableviewmodelselector()</code> plugin in controller.